### DISCUSSION TITLE
Permission denied when installing imageMagick on Mac

### DISCUSSION THREAD

**Discussion by aaron-KO** [ImageMagick/ImageMagick#4901] [Help] [answered]
> **==> make install**
> 
> Last 15 lines from /Users/v96/Library/Logs/Homebrew/ghostscript/02.make:
> 
> ./obj/aux/echogs -e .dev -w- -l-obj ./obj/zlibd_1 -lib z
> 
> make: ./obj/aux/echogs: Permission denied
> 
> ./obj/aux/echogs -e .dev -w- -l-obj ./obj/fapiu
> 
> cp ./base/gsftopts.h ./obj/gsftopts.h
> 
> make: *** [obj/zlibd_1.dev] Error 1
> 
> make: *** Waiting for unfinished jobs....

it seems like i need to sudo install it, but i don't have sudo access, is there a way to install without sudo?

**Comment by urban-warrior** [comment]
This is an ImageMagick discussion forum.  We can help if you post an issue related to ImageMagick.  Perhaps you want to post your issue to a ghostscript forum instead.

**Reply by aaron-KO** [reply]
oh sorry didn't mention that this was happened when i was installing the ImageMagick

**Reply by dlemstra** [reply]
It looks like you are using homebrew to build ImageMagick and it seems that this builds Ghostscript. I think you will need to ask your question to the homebrew team instead.

**Comment by aaron-KO** [comment, marked as answer]
I resoloved the issue by unziped the dependency and reconfigured them and make install

**Reply by PHc92** [reply]
@aaron-KO I am facing the exact same issue. Could you please explain which dependency you unzipped and what reconfiguration you did?

I tried unzipping ghostpdl and reconfigured with " --with-system-libtiff" but that did not work.

**Comment by fmw42** [comment]
Have you tried installing from Homebrew?

**Reply by PHc92** [reply]
Yes

**Reply by fmw42** [reply]
The way I install on my Mac M1 OSX Monterey is to install all my delegates from MacPorts, then install Imagemagick from source. See my old posts at https://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=29100 and https://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=21502&p=88202&hilit=OSX+MacPorts#p88202

My current config command is:

```
./configure \
--enable-openmp \
--disable-opencl \
--enable-delegate-build --enable-shared --disable-static \
--with-modules --with-quantum-depth=16 --without-wmf --with-rsvg --with-fftw \
--disable-silent-rules --disable-dependency-tracking --without-pango \
--with-lqr --with-gslib --with-gs-font-dir=/opt/local/share/ghostscript/fonts/

```

The issue I have is the Mac default compiler does not seem to install OpenMP or OpenCL. It appears that Apple has made compiling OpenCL and OpenMP much harder that just finding a different compiler. I believe that it requires llvm and then some custom commands.

**Reply by fmw42** [reply]
What happens with the Homebrew install?
